src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Tables/DeltaTable.cs

```diff
@@ -31,6 +31,42 @@ internal DeltaTable(JvmObjectReference jvmObject)
 
         JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
 
+        /// <summary>
+        /// Create a DeltaTable from the given parquet table and partition schema.
+        /// Takes an existing parquet table and constructs a delta transaction log in the base path
+        /// of that table.
+        /// 
+        /// Note: Any changes to the table during the conversion process may not result in a
+        /// consistent state at the end of the conversion. Users should stop any changes to the
+        /// table before the conversion is started.
+        /// 
+        /// An example usage would be
+        /// <code>
+        /// DeltaTable.ConvertToDelta(
+        ///     spark,
+        ///     "parquet.`/path`",
+        ///     var partitionSchema = new StructType(new List&lt;StructField&gt;() {
+        ///         new StructField("key1", new LongType()),
+        ///         new StructField("key2", new StringType())
+        ///     });
+        /// </code>
+        /// </summary>
+        /// <param name="spark">The relevant session.</param>
+        /// <param name="identifier">String used to identify the parquet table.</param>
+        /// <param name="partitionSchema">Struct representing the partition schema.</param>
+        /// <returns>The converted DeltaTable.</returns>
+        public static DeltaTable ConvertToDelta(
+            SparkSession spark,
+            string identifier,
+            StructType partitionSchema) =>
+            new DeltaTable(
+                (JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+                s_deltaTableClassName,
+                "convertToDelta",
+                spark,
+                identifier,
+                partitionSchema));
```

Comment:
I don't think this is correct. Check https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Sql/DataFrameReader.cs#L51 to see how `StructType` is used.

This is causing the test failure: https://dev.azure.com/dnceng/public/_build/results?buildId=460232&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774&t=4953245d-467f-515f-f1c4-921c8183805f

Reply:
I see. That makes sense. I have made changes to test this. Thanks for pointing out this!